extensions/AtlasMongoDb/AtlasMongoDb/MongoDbKernelMemoryStorage.cs

```diff
@@ -0,0 +1,216 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.KernelMemory.ContentStorage;
+using MongoDB.Bson;
+using MongoDB.Driver;
+using MongoDB.Driver.GridFS;
+
+namespace Alkampfer.KernelMemory.AtlasMongoDb;
+
+public class MongoDbKernelMemoryStorage : MongoDbKernelMemoryBaseStorage, IContentStorage
+{
+    public MongoDbKernelMemoryStorage(MongoDbKernelMemoryConfiguration config) : base(config)
+    {
+    }
+
+    public Task CreateIndexDirectoryAsync(string index, CancellationToken cancellationToken = new CancellationToken())
+    {
+        return Task.CompletedTask;
+    }
+
+    public async Task DeleteIndexDirectoryAsync(string index, CancellationToken cancellationToken = new CancellationToken())
+    {
+        // delete all document in gridfs that have index as metadata
+        var filter = Builders<GridFSFileInfo>.Filter.Eq("metadata.index", index);
+        // load all id then delete all id
+        var bucket = this.GetBucketForIndex(index);
+        await bucket.DropAsync(cancellationToken).ConfigureAwait(false);
+
+        await this.Database.DropCollectionAsync(index, cancellationToken).ConfigureAwait(false);
+    }
+
+    public Task CreateDocumentDirectoryAsync(string index, string documentId,
+        CancellationToken cancellationToken = new CancellationToken())
+    {
+        //no need to create anything for the document
+        return Task.CompletedTask;
+    }
+
+    private static async Task<IAsyncCursor<GridFSFileInfo<string>>> GetFromBucketByIdAsync(string id, GridFSBucket<string> bucket, CancellationToken cancellationToken)
```

Comment:
here and in other files, could you sort methods by visibility?
This is the order we try to keep across files (with few exceptions):
* public constants values / static values (possibly no public constants, they are all grouped in one file)
* private constants / static values
* public getters/setters
* ctors
* public methods
* public static methods
* private methods
* private static methods

Reply:
I should have fixed ordering.